### PR TITLE
Add shared lib fn to get csproj version

### PR DESCRIPTION
**Pull request by jaucourt**
https://eaflood.atlassian.net/browse/PSD-472

Shared library needs function to extract version from .csproj

### CODE REVIEW

File: src/uk/gov/defra/ffc/DefraUtils.groovy [hunk 1]
```diff
@@ -7,6 +7,14 @@ def repoUrl = ''
 def commitSha = ''
 def workspace
 
+def getCSProjVersion(projName) {
+  def version = sh(returnStdout: true, script: "xmllint $projName/$projName\.csproj --xpath '//Project/PropertyGroup/Version/text()'").trim()
+  if (version == '') {
+    version = sh(returnStdout: true, script: "xmllint $projName/$projName\.csproj --xpath 'concat(//Project/PropertyGroup/VersionPrefix/text(), \"-\", //Project/PropertyGroup/VersionSuffix/text())'").trim()
```
Comment (markysoft) [inline src/uk/gov/defra/ffc/DefraUtils.groovy]: Looks like the version structure used in our project is the `PropertyGroup/Version`, one. 
Could the code be simplified to be the only the first bit of logic if we standardised on how the version is set in the csproj files?

Comment (jaucourt) [reply src/uk/gov/defra/ffc/DefraUtils.groovy]: It absolutely could - I was trying to make it a bit more bulletproof, seemingly there is no guarantee that either Version or VersionPrefix / VersionSuffix will be set, having created a new dotnet core console project and had a look at the csproj file it generated. I'll check with the dot net guys

File: README.md [hunk 2]
```diff
@@ -35,6 +35,10 @@ defraUtils.updateGithubCommitStatus('Build started', 'PENDING')
 ```
 ## Functions
 
+### getCSProjVersion
+
+Returns the project version from the `[projectName].csproj` file
```
Comment (johnwatson484) [inline README.md]: I'd give more detail about how to use this as unlike the Node.js equivalent a parameter is needed.  We should explain what it is.

File: src/uk/gov/defra/ffc/DefraUtils.groovy (hunk 1 above)
Comment (johnwatson484) [reply src/uk/gov/defra/ffc/DefraUtils.groovy]: Here's my comments from Slack so we have a rationale for the decision in one place.

I agree we should settle on a standard of using the `Version`.  We have no need identified for suffixes or prefixes with our version strategy so let's not add complexity we can avoid.  If there is a future need we can revisit.  This standard should also be added to the dev guide but we shouldn't increase the scope of this ticket.  Creating a new ticket for that update is fine.

**Pull request merged**